src/main/java/org/fenixedu/academic/ui/spring/StrutsFunctionalityController.java

```diff
@@ -34,7 +35,10 @@
     @ModelAttribute
     private void selectStrutsFunctionality(HttpServletRequest request) {
         Functionality functionality = RenderersAnnotationProcessor.getFunctionalityForType(getFunctionalityType());
-        BennuPortalDispatcher.selectFunctionality(request,
-                MenuFunctionality.findFunctionality(functionality.getProvider(), functionality.getKey()));
+        MenuFunctionality menuItem = MenuFunctionality.findFunctionality(functionality.getProvider(), functionality.getKey());
+        if (!menuItem.isAvailableForCurrentUser()) {
```

Comment:
You should add a `null` check here, as the selected functionality may not be installed, yielding a `null` Menu Item.